javascripts/tileCollection.js

```diff
@@ -0,0 +1,57 @@
+function TileCollection(direction) {
+  // direction will be keypress values -- 37: left, 39: right, 38: up, or 40: down
+  this.direction = direction; // needed?
+
+  if (direction == 37 || direction == 39) {
+    this.collection_type = 'row';
+    this.attr_name = 'data-col';
+    this.max_length = cols.length;
+    this.generate = generateRow;
+  } else if (direction == 38 || direction == 40) {
+    this.collection_type = 'col';
+    this.attr_name = 'data-row';
+    this.max_length = rows.length;
+    this.generate = generateCol;
+  }
+
+  if (direction == 39 || direction == 40) {
+    this.needs_reverse = true;
+  } else if (direction == 37 || direction == 38) {
+    this.needs_reverse = false;
+  }
+}
+
+TileCollection.prototype.move = function() {
+  for (var i = 0; i < this.max_length; i++) {
+    this.shift_tiles(this.combine(this.generate(i)));
+  }
+}
+
+TileCollection.prototype.combine = function(collection) {
+  if (this.needs_reverse) { collection.reverse(); }
```

Comment:
Is there a better way to do this, without duplicating code? We had it written out as a separate function that was almost identical except it started at the end of array and moved to the beginning, but it seemed redundant.